internal/layer2/announcer.go

```diff
@@ -86,20 +86,17 @@ func (a *Announce) updateInterfaces() {
 				continue
 			}
 		}
-		if ifi.Flags&net.FlagBroadcast != 0 {
-			keepARP[ifi.Index] = true
-		}
 
 		for _, a := range addrs {
 			ipaddr, ok := a.(*net.IPNet)
 			if !ok {
 				continue
 			}
-			if ipaddr.IP.To4() != nil || !ipaddr.IP.IsLinkLocalUnicast() {
-				continue
+			if ipaddr.IP.To4() != nil && (ifi.Flags&net.FlagBroadcast) != 0 {
+				keepARP[ifi.Index] = true
```

Comment:
you can continue here, and no need for the else below, just the if

Reply:
yes @fedepaol , done.